### PR TITLE
Mount /usr/share/ca-certificates in the Azure CCM, update Flatcar to 3227.2.1, enforce NFT backend, and improve Terraform configs

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

This PR brings several changes needed to make Azure work on all operating systems:

* Mount `/usr/share/ca-certificates` in the Azure CCM
  * Similar to #2308, this is needed to make the Azure CCM work on Flatcar
* Update Flatcar to 3227.2.1
  * We were initially using a very old version, so this PR bumps it to the latest. This also helps us to avoid issues described in #2318
* Enforce the NFT iptables backend for Canal on RHEL
  * Similar to Flatcar, RHEL uses the NFT backend by default, so this change ensures we don't run into problems with Calico/Canal autodetection mechanisms
* Use the same image reference and plan for the initial MachineDeployment as for the control plane
  * This is disabled for RHEL because machine-controller v1.43 has a stronger dependency on the RHEL version (it works only with RHEL 8.3)

All those changes were tested as part of #2272.

**What type of PR is this?**

/kind bug

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
- Mount `/usr/share/ca-certificates` to the Azure CCM container to fix CrashLoopBackoff on clusters running Flatcar
- Update Flatcar to 3227.2.1 in the example Terraform configs for Azure
- Enforce the NFT iptables backend for Canal and Calico VXLAN on RHEL
- Use the same image reference and plan for the initial Azure MachineDeployment as for the control plane
```

**Documentation**:
```documentation
NONE
```

### CODE REVIEW

File: examples/terraform/azure/output.tf [hunk 1]
```diff
@@ -91,6 +91,8 @@ output "kubeone_workers" {
           # assignAvailabilitySet = true/false
           securityGroupName = azurerm_network_security_group.sg.name
           assignPublicIP    = true
+          imageReference    = var.os != "rhel" ? var.image_references[var.os].image : null
+          imagePlan         = length(var.image_references[var.os].plan) > 0 && var.os != "rhel" ? var.image_references[var.os].plan[0] : null
```
Comment (kron4eg) [inline examples/terraform/azure/output.tf:95]: I'd suggest to change this in variables.tf instead, like this:

```diff
diff --git examples/terraform/azure/variables.tf examples/terraform/azure/variables.tf
index 2e12c9ef..508ba4dd 100644
--- examples/terraform/azure/variables.tf
+++ examples/terraform/azure/variables.tf
@@ -159,17 +159,8 @@ variable "image_references" {
     }
 
     rhel = {
-      image = {
-        publisher = "RedHat"
-        offer     = "rhel-byos"
-        sku       = "rhel-lvm85"
-        version   = "8.5.20220316"
-      }
-      plan = [{
-        name      = "rhel-lvm85"
-        publisher = "redhat"
-        product   = "rhel-byos"
-      }]
+      image = null
+      plan = null
       ssh_username = "rhel-user"
       worker_os    = "rhel"
     }
```

Comment (xmudrii) [reply examples/terraform/azure/output.tf:95]: This is needed for control plane, so we can't do this.

**Pull request merged**